src/cpp/benders/benders_core/CriterionComputation.cpp

```diff
@@ -16,13 +16,17 @@ void CriterionComputation::ComputeCriterion(
     for (int pattern_index(0); pattern_index < criteria_input_size;
          ++pattern_index) {
         auto pattern_variables_indices = var_indices_[pattern_index];
+        double pattern_value = patterns_values[pattern_index];
+        double criteria_value = criteria[pattern_index];
```

Comment:
```suggestion
        double& pattern_value = patterns_values[pattern_index];
        double& criteria_value = criteria[pattern_index];
```

Reply:
Use non-const reference

Reply:
The purpose of this change is specifically to avoid editing the value inside the loop and prevent cache-invalidation that reduce performances